### PR TITLE
fix: correct test for mutually exclusive feature flags

### LINKED ISSUE #1084: Test for enabling multiple backends is broken
We can only compile to use a single backend so if multiple feature flags are enabled it should error.

https://github.com/noir-lang/noir/blob/98c2cd957c3e60ae94936ba71ca864217001abfd/crates/nargo/src/backends.rs#L13-L15

Here however, we're testing nonexistent feature flags and don't account for the two flavours of plonk.

### PR DESCRIPTION
**Pull request by TomAFrench**
# Related issue(s)

Resolves #1084 

# Description

## Summary of changes

I've updated the checks to match the feature flags which are in use.

I've also removed the default feature flag for the `nargo` package. This should be set in the `nargo_cli` crate (and by any direct consumers of `nargo`)

## Test additions / changes

# Checklist

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` with default settings.
- [x] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this PR to the issue(s) that it resolves.
- [x] I have reviewed the changes on GitHub, line by line.
- [x] I have ensured all changes are covered in the description.

## Documentation needs
- [ ] This PR requires documentation updates when merged.

# Additional context

### CODE REVIEW

**Review by jfecher — APPROVED** [review]
👍

File: crates/nargo/Cargo.toml [hunk 1]
```diff
@@ -22,7 +22,6 @@ aztec_wasm_backend = { optional = true, package = "barretenberg_wasm", git = "ht
 marlin_arkworks_backend = { optional = true, git = "https://github.com/noir-lang/marlin_arkworks_backend", rev = "144378edad821bfaa52bf2cacca8ecc87514a4fc" }
 
 [features]
-default = ["plonk_bn254"]
```
Comment (phated) [inline crates/nargo/Cargo.toml:25]: @TomAFrench Why did you remove this default? Now no backend will be built for nargo without using a feature flag.

**Review by phated — CHANGES_REQUESTED** [review]
Why did you remove the default?

File: crates/nargo/Cargo.toml (hunk 1 above)
Comment (TomAFrench) [reply crates/nargo/Cargo.toml:25]: This is because to be able to build `nargo_cli` for different backends require us to turn off this feature flag. We'd then have to by default turn off this default feature in the workspace so we might as well remove it.

Comment (phated) [reply crates/nargo/Cargo.toml:25]: I think we need to leave a default because building from source (for an end-user) shouldn't **require** a feature flag. If you want to remove the default feature, we should be falling back to the backend instead of requiring one

**Comment by TomAFrench** [comment]
@phated and I came to the conclusion that we should be defining the backends in the `nargo_cli` crate and then passing this through into `nargo`. This is necessary to avoid conflicting backend feature flags when we're using `nargo` as a dependency of `noir-wasm` (as this would rely on the wasm backend)
